src/uint8arrays/from-string.js

```diff
@@ -0,0 +1,24 @@
+'use strict'
+
+const multibase = require('multibase')
+const { names } = require('multibase/src/constants')
+const TextEncoder = require('../text-encoder')
+const utf8Encoder = new TextEncoder('utf8')
+
+function fromString (string, encoding = 'utf8') {
+  if (encoding !== 'utf8') {
+    const base = names[encoding]
+
+    if (!base) {
+      throw new Error('Unknown base')
+    }
+
+    string = `${base.code}${string}`
+
+    return Uint8Array.from(multibase.decode(string))
```

Comment:
We can get rid of the copy when https://github.com/multiformats/js-multibase/pull/63 is merged

Reply:
This whole circular dependence makes me feel uncomfortable:
- It is awkward that you have to prefix string to decode
- Non trivial chunk of code is being pulled in now where in most cases I think you only care about utf-8
- multibase also uses this function which could short circuit (if things are overlooked)

It also doesn't appear like other encodings are used anywhere, so is this really necessary ? If so I think it would be far better to factor out logic behind `multibase.decode(string)` into separate library that just exposes underlying encoders / decoders and share it between multibase and this.

Reply:
You can call .base to get the underlying base encoder without prefix